src/plugins/dataflow-tool/nodes/factories/demo-output-rete-node-factory.tsx

```diff
@@ -0,0 +1,150 @@
+import Rete, { Node, Socket } from "rete";
+import { NodeData } from "rete/types/core/data";
+import { DataflowReteNodeFactory } from "./dataflow-rete-node-factory";
+import { DemoOutputControl } from "../controls/demo-output-control";
+import { DemoOutputValueControl } from "../controls/demo-output-value-control";
+import { DropdownListControl } from "../controls/dropdown-list-control";
+import { MinigraphOptions } from "../dataflow-node-plot";
+import { NodeDemoOutputTypes, NodePlotBlue, NodePlotRed } from "../../model/utilities/node";
+
+const minigraphOptions: Record<string, MinigraphOptions> = {
+  "speed": {
+    backgroundColor: NodePlotBlue,
+    borderColor: NodePlotBlue
+  },
+  "tilt": {
+    backgroundColor: "#fff",
+    borderColor: NodePlotRed
+  }
+};
+
+export class DemoOutputReteNodeFactory extends DataflowReteNodeFactory {
+  constructor(numSocket: Socket) {
+    super("Demo Output", numSocket);
+  }
+
+  public builder(node: Node) {
+    super.defaultBuilder(node);
+    if (this.editor) {
+      node
+        .addControl(new DropdownListControl(this.editor, "outputType", node, NodeDemoOutputTypes, true))
+        .addControl(new DemoOutputControl(this.editor, "demoOutput", node));
+
+      this.addInput(node, "nodeValue");
+
+      if (node.data.outputType === "Grabber") {
+        this.setupGrabberInputs(node);
+      }
+
+      return node as any;
+    }
+  }
+
+  public worker(node: NodeData, inputs: any, outputs: any) {
+    const n1 = inputs.nodeValue.length ? inputs.nodeValue[0] : node.data.nodeValue;
+    // if there is not a valid input, use 0
+    // otherwise convert all non-zero to 1
+    const result = isNaN(n1) ? 0 : +(n1 !== 0);
+    if (this.editor) {
+      const _node = this.editor.nodes.find((n: { id: any; }) => n.id === node.id);
+      if (_node) {
+        const outputTypeControl = _node.controls.get("outputType") as DropdownListControl;
+        const outputType = outputTypeControl.getValue();
+
+        // Update main display
+        const nodeValue = _node.inputs.get("nodeValue")?.control as DemoOutputValueControl;
+        nodeValue?.setValue(result);
+        if (outputType === "Light Bulb") {
+          nodeValue?.setDisplayMessage(result === 0 ? "off" : "on");
+        } else {
+          nodeValue?.setDisplayMessage(result === 0 ? "closed" : "open");
+        }
```

Comment:
At some point someone is going to want to add another output option to this block, so we'll probably want to have properties like this configured via a configuration object rather than be hard-coded, but we can wait for that next request to worry about it. 😉